### PR TITLE
Introduce str_remove_inside and str_remove_emoji functions for text wrangling.

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

Introduce str_remove_inside and str_remove_emoji function for text wrangling.

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [x] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/string_operation.R [hunk 1]
```diff
@@ -615,7 +615,58 @@ str_extract_inside <- function(column, begin = "(", end = ")") {
...
+  if(stringr::str_length(end) > 1) {
+    stop("The end argument must be one character.")
+  }
+  if(grepl("[A-Za-z]", begin)) {
+    stop("The begin argument must be symbol such as (, {, [.")
+  }
+  if(grepl("[A-Za-z]", end)) {
+    stop("The end argument must be symbol such as ), }, ].")
+  }
+  exp = stringr::str_c("\\", begin, "[^()]*\\", end)
```
Comment (kei51e) [inline R/string_operation.R:640]: This one fails to parse the following case. 

```
> str_remove_inside("abc[12(34)56]ijk", begin="[", end="]")
[1] "abc[12(34)56]ijk"
```

This should look like this.

```
  exp <- stringr::str_c("\\", begin, "[^\\" , begin, "\\", end, "]*\\", end)

```

Comment (kei51e) [reply R/string_operation.R:640]: Also better using '<-' for assignment.

File: R/string_operation.R [hunk 2]
```diff
@@ -615,7 +615,58 @@ str_extract_inside <- function(column, begin = "(", end = ")") {
     stop("The end argument must be symbol such as ), }, ].")
   }
   exp = stringr::str_c("(?<=\\", begin, ").*?(?=\\", end, ")")
```
Comment (kei51e) [inline R/string_operation.R:617]:
```suggestion
  exp <- stringr::str_c("(?<=\\", begin, ").*?(?=\\", end, ")")
```

File: R/string_operation.R (hunk 1 above)
Comment (hidekoji) [reply R/string_operation.R]: Updated as suggested.

**Review by kei51e — APPROVED** [review]
Looks good!

**Pull request merged**